code/modules/shuttle/marine_dropship.dm

```diff
@@ -976,48 +1000,123 @@
 	return ..()
 
 
-/obj/machinery/computer/shuttle/shuttle_control/ui_interact(mob/user)
-	if(!allowed(user))
-		to_chat(user, "<span class='warning'>Access Denied!</span>")
+/obj/machinery/computer/shuttle/shuttle_control/ui_interact(mob/user, datum/tgui/ui)
+	if(!(SSshuttle.getShuttle(shuttleId)))
+		RelinkShuttleId()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "ShuttleControl")
+		ui.open()
+
+/obj/machinery/computer/shuttle/shuttle_control/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
 		return
+
+	if(isxeno(usr))
+		return
+
+	if(!allowed(usr))
+		to_chat(usr, "<span class='danger'>Access denied.</span>")
+
+	switch(action)
+		if("selectDestination")
```

Comment:
you can probably early return this, if only one action is possible (so no switch)